source/detail/platform_abstraction.cpp

```diff
@@ -650,26 +640,6 @@ namespace nana
 	}
 #endif
 
-
-	void platform_abstraction::initialize()
-	{
-		if (nullptr == data::storage)
-			data::storage = new platform_runtime;
-	}
-
-	void platform_abstraction::shutdown()
-	{
-		auto & r = platform_storage();
```

Comment:
A check is needed before destroying the `platform_runtime`. Otherwise, when the program exists, it generates a incomprehensible crash if there is still a font object not releasing

Reply:
What we should do if the check fails? Previously an exception was thrown ... but throwing in `platform_abstraction::shutdown()` which is called in `platform_spec` destructor is also a crash.

It would be better to guuarantee that all objects which used `font` are destroyed before `platform_runtime` is destroyed.